packages/iTwinUI-react/src/core/utils/components/Resizer.test.tsx

```diff
@@ -0,0 +1,368 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import { fireEvent, render } from '@testing-library/react';
+import Resizer, { ResizerProps } from './Resizer';
+import * as Styles from '../functions/styles';
+
+jest.spyOn(Styles, 'getTranslateValues').mockReturnValue([100, 100]);
+jest.spyOn(window, 'getComputedStyle').mockReturnValue({
+  minWidth: '50px',
+  minHeight: '50px',
+} as CSSStyleDeclaration);
+jest.spyOn(HTMLElement.prototype, 'getBoundingClientRect').mockReturnValue({
+  width: 100,
+  height: 100,
+} as DOMRect);
+
+const setStyle = jest.fn();
+
+const TestComponent = (props?: Partial<ResizerProps>) => {
+  const ref = React.useRef<HTMLDivElement>(null);
+  return (
+    <div
+      id='test-component'
+      ref={ref}
+      style={{ width: 100, height: 100, transform: 'translate(100px, 100px)' }}
+    >
+      <Resizer
+        elementRef={ref}
+        containerRef={{
+          current: {
+            getBoundingClientRect: () => ({
+              left: 0,
+              right: 300,
+              top: 0,
+              bottom: 300,
+            }),
+          } as HTMLElement,
+        }}
+        setStyle={setStyle}
+        {...props}
+      />
+    </div>
+  );
+};
+
+const renderComponent = (props?: Partial<ResizerProps>) => {
+  return render(<TestComponent {...props} />);
+};
+
+afterEach(() => {
+  jest.clearAllMocks();
+});
+
+/*
+ * For better understanding how everything is positioned.
+ * +------------------------------+
+ * |                              |
+ * |  (100;100)                   |
+ * |      +---------------+       |
+ * |      |    w=100px    |       |
+ * |      |               |       |
+ * |      |               |       |
+ * |      | h=100px       |       | h=300px
+ * |      |               |       |
+ * |      |    element    |       |
+ * |      +---------------+       |
+ * |                  (200;200)   |
+ * |                              |
+ * |         container            |
+ * +------------------------------+
+ *            w=300px         (300;300)
+ */
+
```

Comment:
Thought of using `it.each` but it looked that it will take almost same amount of code and might not be that readable.